### PR TITLE
feat: use the versions from overrides when adding deps without specs

### LINKED ISSUE #4313: overrides doesn't work as expected
### Code to reproduce the issue:

Just a package.json like this:

```json
{
  "name": "test-pnpm",
  "version": "1.0.0",
  "description": "",
  "main": "index.js",
  "scripts": {
    "test": "echo \"Error: no test specified\" && exit 1"
  },
  "author": "",
  "license": "ISC",
  "pnpm": {
    "overrides": {
      "lz-string": "github:pieroxy/lz-string#2f749bc9fe3cc889fa9e32d2769a930d977596e6"
    }
  }
}
```

Then run `pnpm add lz-string`

### Expected behavior:

Updated package.json like this:

```json5
{
  // ...
  "dependencies": {
    "lz-string": "github:pieroxy/lz-string#2f749bc9fe3cc889fa9e32d2769a930d977596e6"
  }
  // ...
}
```

### Actual behavior:

Overrides don't have effect when add a new direct dependency, and the package.json become:

```json5
{
  // ...
  "dependencies": {
    "lz-string": "^1.4.4" // latest version
  }
  // ...
}
```

After running `pnpm add lz-string` again, then it will be corrected.

### Additional information:

 - `node -v` prints: 14.17.2
 - Windows, macOS, or Linux?: macOS

### PR DESCRIPTION
**Pull request by zkochan**
close #4313

### CODE REVIEW

File: .changeset/wicked-sheep-tell.md [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+---
+
+When adding a new dependency, use the version specifier from the overrides, when present [#4313](https://github.com/pnpm/pnpm/issues/4313).
+
+Normally, if the latest version of `foo` is `2.0.0`, then `pnpm add foo` installs `foo@^2.0.0`. This behavior changes if `foo` is specified in an override:
+
+```json
+{
+  "pnpm": {
+    "foo": "1.0.0"
```
Comment (kenrick95) [inline .changeset/wicked-sheep-tell.md]: did you mean

```
"pnpm": {
  "overrides": {
    "foo": "1.0.0"
  }
}
```

Comment (zkochan) [reply .changeset/wicked-sheep-tell.md]: yes, you are right. Fixed.

**Review by kenrick95 — APPROVED** [review]

**Comment by nikoladev** [comment]
In the changeset description there's this part:

> However, if a version is explicitly specifying, then the specified version will be used and the override will be ignored. So `pnpm add foo@0` will install v0 and it doesn't matter what is in the overrides.

Shouldn't there also be a test that ensures that `pnpm add foo@0` will install v0 even if there is an override? So that we don't accidentally break this behavior.

Or does a test exist for this in some other file that I've missed? If not, I'd like to open up a PR for this.

**Comment by zkochan** [comment]
there is no test for it.

**Comment by nikoladev** [comment]
Cool, I'll add it now

**Pull request merged**